File: packages/core/src/engine/verbs/impute.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */

import type ColumnTable from 'arquero/dist/types/table/column-table'
import type { TableStore } from '../../index.js'
import type { FillArgs, Step } from '../../types.js'
import type { ExprFunctionMap } from './types.js'

/**
 * Executes an arquero impute
 * @param step
 * @param store
 * @returns
 */
export async function impute(
	step: Step,
	store: TableStore,
): Promise<ColumnTable> {
	const { input, args } = step
	const { value, to } = args as FillArgs
	const inputTable = await store.get(input)

	const dArgs: ExprFunctionMap = {
		[to]: (_d: any, $: any) => $.value,
	}

	return inputTable.params({ value }).impute(dArgs)
}

File: python/data-wrangling-components/data_wrangling_components/engine/verbs/impute.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#

from typing import List, Union

from data_wrangling_components.table_store import TableContainer


def impute(
    input: TableContainer, columns: List[str], value: Union[str, int, float, bool]
):
    input_table = input.table
    output = input_table.copy()
    for column in columns:
        output[column] = output[column].fillna(value)
    return TableContainer(table=output)
